### PR TITLE
Fix all remaining unit tests under Python 3

### PR DESCRIPTION
**Pull request by mvdbeek**
xref. #1715

### CODE REVIEW

**Comment by mvdbeek** [comment]
I think all unit tests should be passing now, and I even managed to run a couple of tools successfully!

File: lib/galaxy/jobs/__init__.py [hunk 1]
```diff
@@ -1546,7 +1546,7 @@ def check_limits(self, runtime=None):
         return None
 
     def has_limits(self):
-        has_output_limit = self.app.job_config.limits.output_size > 0
+        has_output_limit = self.app.job_config.limits.output_size and self.app.job_config.limits.output_size > 0
```
Comment (nsoranzo) [inline lib/galaxy/jobs/__init__.py:1549]: Need to do the same at line 1533

**Comment by mvdbeek** [comment]
Hmm, the failing tests in https://travis-ci.org/mvdbeek/galaxy/jobs/399571214 pass fine locally, but I'm running 3.6 instead of 3.4 ... do you think it's worth switching travis to 3.6 ?

File: lib/galaxy/jobs/__init__.py (hunk 1 above)
Comment (mvdbeek) [reply lib/galaxy/jobs/__init__.py:1549]: Good catch, done that in aaf968c

File: lib/galaxy_ext/metadata/set_metadata.py [hunk 2]
```diff
@@ -26,7 +26,7 @@
 from galaxy.util import stringify_dictionary_keys
 
 # ensure supported version
-assert sys.version_info[:2] >= (2, 6) and sys.version_info[:2] <= (2, 7), 'Python version must be 2.6 or 2.7, this is: %s' % sys.version
+assert sys.version_info[:2] >= (2, 7) and sys.version_info[:2] <= (3, 7), 'Python version must be at least 2.7, this is: %s' % sys.version
```
Comment (nsoranzo) [inline lib/galaxy_ext/metadata/set_metadata.py]: Can we just drop the `sys.version_info[:2] <= ()` ?

**Comment by nsoranzo** [comment]
@natefoo Can you help with the last remaining [unit test error](https://travis-ci.org/galaxyproject/galaxy/jobs/399960334) which involves `lib/galaxy/containers/docker_swarm.py` ?

**Review by nsoranzo — APPROVED** [review]

**Comment by nsoranzo** [comment]
All green!

File: .travis.yml [hunk 3]
```diff
@@ -34,7 +34,6 @@ matrix:
     - env: TOX_ENV=py34-unit
       addons: *py3_addons
   allow_failures:
-    - env: TOX_ENV=py34-unit
```
Comment (martenson) [inline .travis.yml:37]: 🎉

**Pull request merged**